REACT_STYLEGUIDE.MD

```diff
@@ -0,0 +1,194 @@
+# React Style Guide
+
+### Prefer Stateless functional components where possible.
+Stateless function components are more concise, and there are plans for react to increase performance of them.
+Good:
+```
+export function KuiButton({ onClick, isDisabled }) {
+  return <button className="kuiButton" onClick={onClick} isDisabled={isDisabled}/>
+};
+```
+Bad:
+```
+export class KuiButton extends React.Component {
+  render({ onClick, isDisabled }) {
+    return <button className="kuiButton" onClick={onClick} isDisabled={isDisabled}/>
+  }
+}
+```
+
+### When state is involved, use ES6 style React Classes over ES5.
+Good:
+```
+export class ClickCounter extends React.Component {
+  constructor() {
+    this.state = {
+      clickCount: 0
+    };
+  }
+
+  onClick() {
+    this.setState({ clickCount: this.state.clickCount++ });
+  }
+
+  render({ onClick }) {
+    return <button className="kuiButton" onClick={this.onClick}/>
+  }
+}
+```
+Bad:
+```
+ export const ClickCounter = React.createClass({
+  getInitialState() {
+    return {
+      clickCount: 0
+    };
+  },
+  onClick() {
+    this.setState({ clickCount: this.state.clickCount++ });
+  },
+  render({ onClick }) {
+    return <button className="kuiButton" onClick={this.onClick}/>
+  }
+});
+```
+
+### Prefer reactDirective over react-component
+reactDirective and react-component are two different ways of embedding react in angular. Using `react-component` means adding a bunch of components into angular, while `reactDirective` keeps them isolated, and is also a more sucinct syntax.
+
+Good:
+```
+<hello-component fname="person.fname" lname="person.lname" watch-depth="reference"></hello-component>
+```
+Bad:
+```
+<react-component name="HelloComponent" props="person" watch-depth="reference"/>
+```
+
+### Prefix ui_framework elements with kui, but not their file names.
+Good:
+```
+button.js:
+export function KuiButton({ onClick, isDisabled }) {
+  return <button className="kuiButton" onClick={onClick} isDisabled={isDisabled}/>
+};
+```
+Bad:
+```
+button.js:
+export class Button extends React.Component {
+  render({ onClick, isDisabled }) {
+    return <button className="kuiButton" onClick={onClick} isDisabled={isDisabled}/>
+  }
+}
+```
+The filenames leave it off because snake casing already increases file name length. The prefix is left on for the components to avoid any naming collisions.
```

Comment:
I don't mind keeping this rule but I think the real reason we leave off `kui_` from the name is simply inertia... none of the files have it, so it would be a (small) chore to update them all and update the import references, and I think it's not a very pressing issue.